pkg/operator/ceph/controller/spec.go

```diff
@@ -124,6 +126,16 @@ if [ "$LOG_MAX_SIZE" != "0" ]; then
 fi
 
 while true; do
+	if [ "$MAX_COREDUMP_KEEP" != 0 ]; then
+		# Keep at most MAX_COREDUMP_KEEP core dump files
+		nsenter -t 1 -m -- su -c "ls -1tr /var/log/ceph/core.* | head -n -${MAX_COREDUMP_KEEP} | xargs -d '\n' rm -f --"
+	fi
+
+	if [ "$MAX_COREDUMP_AGE_DAYS" != 0 ]; then
+		# Keep core dump files for MAX_COREDUMP_AGE_DAYS days
+		nsenter -t 1 -m -- su -c "find /var/log/ceph -name 'core.*' -mtime +${MAX_COREDUMP_AGE_DAYS} -delete"
+	fi
+
```

Comment:
IIRC, the issue mentioned you can update logrotate config to implement it.

Reply:
@sp98 thinks different configurations for different types of logs files (*.logs and core.PID) will be harder to implement and manage. 
It is more efficient and simpler to write shell script directly instead of using logratate.

Reply:
He means to use same logrotate configurations for both *.logs and core.PID files.

Reply:
Yes. I meant to use the same logrotate config for both file types. Something like: 
``` 
/var/log/ceph/core.*
/var/log/ceph/*.log {
    rotate 7
    daily
    compress
    sharedscripts
    postrotate
        killall -q -1 ceph-mon ceph-mgr ceph-mds ceph-osd ceph-fuse radosgw rbd-mirror cephfs-mirror || pkill -1 -x "ceph-mon|ceph-mgr|ceph-mds|ceph-osd|ceph-fuse|radosgw|rbd-mirror|cephfs-mirror" || true
    endscript
    missingok
    notifempty
    su root ceph
}```

But @thenamehasbeentake suggested that configuration for `*.log` and `core.*` needed to be different.

Reply:
> Yes. I meant to use the same logrotate config for both file types. Something like:
> 
> ```
> /var/log/ceph/core.*
> /var/log/ceph/*.log {
>     rotate 7
>     daily
>     compress
>     sharedscripts
>     postrotate
>         killall -q -1 ceph-mon ceph-mgr ceph-mds ceph-osd ceph-fuse radosgw rbd-mirror cephfs-mirror || pkill -1 -x "ceph-mon|ceph-mgr|ceph-mds|ceph-osd|ceph-fuse|radosgw|rbd-mirror|cephfs-mirror" || true
>     endscript
>     missingok
>     notifempty
>     su root ceph
> }
> ```
> 
> But @thenamehasbeentake suggested that configuration for `*.log` and `core.*` needed to be different.

With such logrote.conf configuration, the core file will not be deleted. Each rotation will cause core file named like core.123 to become core.123.1, and the new core file named core.123.1 can still match core.*.